src/webots/control/WbLanguageTools.cpp

```diff
@@ -107,15 +108,19 @@ QString WbLanguageTools::pythonCommand(QString &shortVersion, const QString &com
     shortVersion = checkIfPythonCommandExist(pythonCommand, env, true);
   else {
     if (pythonCommand == "python" || pythonCommand == "python3") {
-      pythonCommand = findWorkingPythonPath("3.8", env, false);
-      shortVersion = "38";
+      pythonCommand = findWorkingPythonPath("3.10", env, false);
+      shortVersion = "310";
```

Comment:
We should not assign the `shortVersion` unless we know the python command is valid (e.g., `!= "!"`) otherwise, it will be assigned several times (see my suggestion below).

Reply:
This change sounds good, done.